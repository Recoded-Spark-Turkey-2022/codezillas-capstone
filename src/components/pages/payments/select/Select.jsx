/* eslint-disable no-unused-expressions */
/* eslint-disable react/destructuring-assignment */
export default function Select(props)
{
    const {options,defaultVal} = props;
    function handleSelect(e)
    {
        props.func ? props.func(e.target.selectedOptions[0].value,undefined) : ''
        props.selected ? props.selected(e.target.selectedOptions[0].value) : ''
          
    }
    return(
        <select 
        className={props.AllClassess}
        onChange={(e) => handleSelect(e)   }
        >
            <option selected="selected">{defaultVal}</option>
            {options?.map(element => {
                return <option key={element?.country_name} value={element?.country_name || element?.state_name}>{element?.country_name || element?.state_name}</option>
            })}
        </select>
    )
}