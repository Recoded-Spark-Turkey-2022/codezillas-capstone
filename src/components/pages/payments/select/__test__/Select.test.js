import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Select from '../Select';
import { store } from '../../../../../app/store';


test('Navbar component renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Select />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
