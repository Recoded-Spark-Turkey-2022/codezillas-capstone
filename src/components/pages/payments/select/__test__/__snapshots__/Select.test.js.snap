// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders correctly 1`] = `
<select
  onChange={[Function]}
>
  <option
    selected="selected"
  />
</select>
`;
